kphp_polyfills.php

```diff
@@ -1286,4 +1286,132 @@ function ffi_array_get(\FFI\CData $arr, int $index) {
 #endregion
 
 
+#region kphp_tracing
+
+// KPHP tracing is a technology that aims to collect richer variety of data compared to Open telemetry
+// and faster compared to existing tracing SDKs.
+// In fact, every single request is traced in-memory with zero overhead, and probably flushed to disk on finish
+// (if something interesting happens during a script execution, we have its trace — for any given request).
+//
+// KPHP tracing is not implemented for PHP at all (and has no reason to be).
+// In PHP, all tracing functions do nothing.
+
+/**
+ * "Div" represents a piece of a distributed Trace written by a single process handling one request.
+ * It's a set of spans (@see KphpSpan below).
+ * In case of KPHP, when a worker starts handling a PHP request, it starts a div,
+ * which is finished on php script finish (no watter whether it's a general/job/rpc worker).
+ * A term "Div" was invented specially, not to be confused with a word "Trace":
+ * Trace is a set of Div with equal TraceID, it's a distributed entity,
+ * whereas Div starts and finishes by a single trace-emitting node.
+ */
+final class KphpDiv {
+  /**
+   * @return tuple(int, int)
+   */
+  public function generateTraceCtxForChild(int $div_id, int $trace_flags) {
+    return tuple(0, 0);
+  }
+
+  public function assignTraceCtx(int $int1, int $int2, int $override_div_id): int {
+    return 0;
+  }
+
+  public function getStartTimestamp(): float {
+    return 0.0;
+  }
+
+  public function getEndTimestamp(): float {
+    return 0.0;
+  }
+}
+
+/**
+ * Span is an atomic element inside a Div.
+ * As opposed to Open telemetry, KPHP does not store any state of spans: it does not have a Span class
+ * containing start/finish timestamps, name, parent, children, attributes, links, etc.
+ * Instead, when any event occurs, it is written to a binary in-memory append-only log (kphp_tracing_binlog.h).
+ * That binlog is a compact representation throughout all trace timeline.
+ *
+ * As a consequence, addAttribute() exists, but getAttribute() does not.
+ * updateTitle() exists, but getTitle() does not. And so on.
```

Comment:
was `updateTitle()` this function given as an example, or did you mean `updateName()`?

Reply:
I've already changed this before your comment)